tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/DotNetLanguageService.Checks.cs

```diff
@@ -2,154 +2,130 @@
 using Azure.Sdk.Tools.Cli.Models;
 using Azure.Sdk.Tools.Cli.Models.Responses.Package;
 
-namespace Azure.Sdk.Tools.Cli.Services;
+namespace Azure.Sdk.Tools.Cli.Services.Languages;
 
 /// <summary>
 /// .NET-specific implementation of language repository service.
 /// Uses tools like dotnet CLI, MSBuild, NuGet, etc. for .NET development workflows.
 /// </summary>
-public class DotNetLanguageSpecificChecks : ILanguageSpecificChecks
-{
-    private readonly IProcessHelper _processHelper;
-    private readonly IGitHelper _gitHelper;
-    private readonly IPowershellHelper _powershellHelper;
-    private readonly ILogger<DotNetLanguageSpecificChecks> _logger;
-    private readonly ICommonValidationHelpers _commonValidationHelpers;
-    private const string DotNetCommand = "dotnet";
-    private const string RequiredDotNetVersion = "9.0.102"; // TODO - centralize this as part of env setup tool
-    private static readonly TimeSpan CodeChecksTimeout = TimeSpan.FromMinutes(6);
-    private static readonly TimeSpan AotCompatTimeout = TimeSpan.FromMinutes(5);
-
-    public DotNetLanguageSpecificChecks(
-        IProcessHelper processHelper,
-        IPowershellHelper powershellHelper,
-        IGitHelper gitHelper,
-        ILogger<DotNetLanguageSpecificChecks> logger,
-        ICommonValidationHelpers commonValidationHelpers)
-    {
-        _processHelper = processHelper;
-        _powershellHelper = powershellHelper;
-        _gitHelper = gitHelper;
-        _logger = logger;
-        _commonValidationHelpers = commonValidationHelpers;
-    }
-
-    public async Task<PackageCheckResponse> CheckGeneratedCode(string packagePath, bool fixCheckErrors = false, CancellationToken ct = default)
+public partial class DotnetLanguageService : LanguageService
+{ 
+    public override async Task<PackageCheckResponse> CheckGeneratedCode(string packagePath, bool fixCheckErrors = false, CancellationToken ct = default)
     {
         try
         {
-            _logger.LogInformation("Starting generated code checks for .NET project at: {PackagePath}", packagePath);
+            logger.LogInformation("Starting generated code checks for .NET project at: {PackagePath}", packagePath);
```

Comment:
nit: when using class fields folks often use `this.logger` so it is clear where the variable is coming from. However, that is probably a bigger style type clean-up we might consider as a follow-up.